common/persistence/persistenceMetricClients.go

```diff
@@ -250,6 +249,7 @@ func (p *executionPersistenceClient) SetWorkflowExecution(
 ) (*SetWorkflowExecutionResponse, error) {
 	p.metricClient.IncCounter(metrics.PersistenceSetWorkflowExecutionScope, metrics.PersistenceRequests)
 
+	p.emitPerCategoryTaskMetrics(metrics.PersistenceSetWorkflowExecutionScope, &request.SetWorkflowSnapshot.Tasks)
```

Comment:
can you also emit the metrics for `CreateWorkflowExecution` & `ConflictResolveWorkflowExecution`?

Reply:
I am not sure if persistence metrics client is the right place for emitting this metric. Maybe it should be emitted at the same place as the state transition count in workflow context? 
https://github.com/temporalio/temporal/blob/master/service/history/workflow/context.go#L932

Reply:
Wenquan suggested a similar place - I'll move them. Thanks!